### PR TITLE
[docs] migrate AIR API references

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

This PR disbands the AIR API References page. Most API references were already defined within their individual libraries.

Additional changes include:
1. Move `Preprocessors` to Data.
2. Move Wandb/MLflow/Comet `LoggerCallback`s to Tune.

## Related issue number

#38365

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by amogkam** [comment]
Are we removing all the non-tabular preprocessor API references from the documentation?

File: doc/source/ray-air/api/configs.rst [hunk 1]
```diff
@@ -1,25 +0,0 @@
-
-Ray AIR Configurations
-======================
```
Comment (richardliaw) [inline doc/source/ray-air/api/configs.rst:3]: assuming this already got moved to train?

**Review by richardliaw — APPROVED** [review]
thanks!

File: doc/source/ray-air/api/configs.rst (hunk 1 above)
Comment (matthewdeng) [reply doc/source/ray-air/api/configs.rst:3]: Yep, the configs are in the Train/Tune API references already.

File: python/ray/tune/logger/comet.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from ray.air.integrations.comet import CometLoggerCallback
+
+CometLoggerCallback.__module__ = "ray.tune.logger.comet"
```
[the remark this answers is not in the document]
Comment (matthewdeng) [reply python/ray/tune/logger/comet.py:3]: This just makes the path show up properly
```python
>>> from ray.tune.logger.comet import CometLoggerCallback
>>> CometLoggerCallback
<class 'ray.tune.logger.comet.CometLoggerCallback'>
```

**Review by amogkam — APPROVED** [review]

**Pull request merged**